File: server/Models/Player.mjs
export class Player {
    constructor(id, hwid, name) {
        this.id = id;
        this.hwid = hwid;
        this.nickname = name;
        this.score = 0;
        this.isDrawing = false;
    }

    // Set nickname
    setNickname(nickname) {
        this.nickname = nickname;
    }

    // Update score
    setScore(score) {
        this.score = score;
    }

    // Update score
    getScore() {
        return this.score;
    }

    getIsDrawing() {
        return this.isDrawing;
    }

    setIsDrawing(isDrawing) {
        this.isDrawing = isDrawing;
    }
}

File: android/app/src/main/java/com/groupfive/sketchmatch/models/Player.kt
package com.groupfive.sketchmatch.models

data class Player(
    var id: String = "",
    var hwid: String = "",
    var nickname: String = "",
    var score: Int = 0,
    var isDrawing: Boolean = false,
    var hasGuessedCorrectly: Boolean = false
)
